E-commerceapp with backend

UtpalTheDev opened the pull request:
This is my e-commerce apps first version. I need suggestions from all of you to make it better, please review it and give suggestions.

UtpalTheDev committed:
Add files via upload

direshpradhan commented on frontend/src/Cart.js:
```diff
@@ -0,0 +1,178 @@
+                    {(item.price - (item.price * item.offer) / 100).toFixed(2)}
+                  </div>
+
+                  <div className="added_to_cart">
+                    <button
+                      onClick={() => {
+                        if (item.qty === 1) {
+                          cart_delete_call(
+                            "https://ecomm-demo.utpalpati.repl.co/cart",
+                            { itemId: item._id }
+                          );
+                        } else {
+                          cart_decrease_call(
+                            "https://ecomm-demo.utpalpati.repl.co/cart",
+                            {
+                              items: { itemId: item._id, qty: item.qty - 1 }
+                            }
+                          );
+                        }
+                      }}
+                    >
+                      -
+                    </button>
+                    {item.qty}
+
+                    <button
+                      onClick={() => {
+                        cart_increase_call(
+                          "https://ecomm-demo.utpalpati.repl.co/cart",
+                          {
+                            items: { itemId: item._id, qty: item.qty + 1 }
+                          }
+                        );
+                      }}
+                    >
+                      +
+                    </button>
+                  </div>
```
You can use the Add_to_cart_button function written in app.js file here.

UtpalTheDev replied:
Thanks for pointing this out. I think this will significantly reduce my lines of code.

direshpradhan commented on frontend/src/App.js:
```diff
@@ -0,0 +1,457 @@
+      <div className="ecom-navbar">
+        <Link to="/">
+          <div className="logo">
+            <img
+              src="http://www.pngmart.com/files/7/Growing-Plant-PNG-Pic.png"
+              alt="logo"
+            />
```
The logo image is not visible.

UtpalTheDev replied:
Thanks for pointing this out. In firefox it was working but in other browsers it is not working. I will fix it.

direshpradhan commented on frontend/src/App.js:
```diff
@@ -0,0 +1,457 @@
+      console.log("err", err);
+    }
+  }
+  function Add_to_cart_button(
+    id,
+    name,
+    image,
+    price,
+    productName,
+    inStock,
+    level,
+    fastDelivery,
+    isnew,
+    ratings,
+    offer
+  ) {
```
Some of the arguments are unused here so you can remove them.

UtpalTheDev replied:
I will remove unused arguments.

direshpradhan reviewed (changes requested):
Hey Utpal, 
The e-commerce app looks great. Just some minor changes in the code can be done which I have suggested.

RawHeatG commented on frontend/src/Products.js:
```diff
@@ -0,0 +1,209 @@
+                  </div>
+                  {isnew ? <div class="cards-badge">New</div> : ""}
+
+                  <div
+                    class="title"
+                    style={{
+                      overflow: "hidden",
+                      textOverflow: "..2.",
+                      height: "20px"
+                    }}
```
Add a little padding, in the title of card as the big names are going till end of card

UtpalTheDev replied:
Thanks for pointing this out. I just missed this part. I will add some padding for the title.

RawHeatG commented on frontend/src/App.js:
```diff
@@ -0,0 +1,457 @@
+    <>
+      <div className="ecom-navbar">
+        <Link to="/">
+          <div className="logo">
+            <img
+              src="http://www.pngmart.com/files/7/Growing-Plant-PNG-Pic.png"
+              alt="logo"
+            />
```
The site name can be aligned with the logo itself, rather than being below the logo, and add a little padding on the left

UtpalTheDev replied:
Thanks for the suggestion. I will change it.

RawHeatG commented on frontend/src/Reducer-context.js:
```diff
@@ -0,0 +1,204 @@
+  }
+}
+
+async function deleting(url, payload) {
+  return await axios.delete(url, payload);
+}
```
Looks like this code isn't being used, if so please remove it

UtpalTheDev replied:
Thanks for pointing this out. I completely missed this part. I will remove this part of code.

RawHeatG reviewed (commented):
The site is awesome, all the states being handled in the backend and design is also neat,
I found some issues here and there, please take a look at them, all in all, Nice work!

UtpalTheDev committed:
refactor: add_to_cart_button function, unused codes + style: logo
1. add_to_cart_button had extra unused arguments.
2. in cart.js replaced button code with add_to_cart_button function.
3. logo not visible and logo positioning not proper.
4. removed unused codes.

UtpalTheDev committed:
fix: products going out of the grid

UtpalTheDev committed:
refactor: logo added

UtpalTheDev committed:
feat: product details page add

UtpalTheDev committed:
refactor: remove unused arguments and variables, fix: link not working in cart and wishlist page.

UtpalTheDev committed:
refactor: remove repeated code

UtpalTheDev committed:
refactor: private route add

UtpalTheDev committed:
feat: auth add

UtpalTheDev committed:
style: form style update

UtpalTheDev committed:
style: login , signup page style change

UtpalTheDev committed:
feat: login error message

UtpalTheDev committed:
refactor: user data call

UtpalTheDev committed:
style: wish button style change, fix: last items not visible in product page
1. overflow handled in wish button in product details page.
2. sort and filter button overlapping with the items card  in mobile version in product page.

UtpalTheDev committed:
Update README.md

UtpalTheDev committed:
fix: asset images not working

UtpalTheDev committed:
refactor: package.json file update

UtpalTheDev committed:
refactor: backend link change

UtpalTheDev committed:
refactor: toast add

UtpalTheDev committed:
refactor: loader add

UtpalTheDev committed:
refactor: style class add

UtpalTheDev committed:
refactor: 404 page add

UtpalTheDev committed:
refactor: style add

UtpalTheDev committed:
refactor: test add

UtpalTheDev committed:
refactor: file structure change

UtpalTheDev committed:
refactor: remove unused imports, console logs.

UtpalTheDev committed:
refactor: backend repository change

UtpalTheDev committed:
fix: images not loading

UtpalTheDev committed:
refactor: title change

UtpalTheDev committed:
refactor: padding add

UtpalTheDev committed:
fix: not returning error message

UtpalTheDev committed:
refactor: component library link change

UtpalTheDev committed:
refactor: move button components to another file

UtpalTheDev committed:
refactor: navbar moved to another file

UtpalTheDev committed:
Merge pull request #2 from UtpalTheDev/Feature

Feature add

UtpalTheDev committed:
refactor: feature add

UtpalTheDev committed:
refactor: payment feature add

UtpalTheDev committed:
Merge pull request #3 from UtpalTheDev/Feature

Payment add

UtpalTheDev committed:
refactor: remove warnings

UtpalTheDev committed:
refactor: manage warnings

UtpalTheDev committed:
refactor: fix error

UtpalTheDev committed:
Merge pull request #5 from UtpalTheDev/Feature

refactor: manage warnings

UtpalTheDev committed:
refactor: illustration add, banner add

UtpalTheDev committed:
refactor: remove comments

UtpalTheDev committed:
refactor: add more description about illustrations

UtpalTheDev committed:
refactor: login form change

UtpalTheDev committed:
Merge pull request #6 from UtpalTheDev/Feature

Illustration add

UtpalTheDev committed:
refactor: illustration add

UtpalTheDev committed:
fix: after checkout cart is not empty

UtpalTheDev committed:
refactor: responsiveness handle, navbar stick

UtpalTheDev committed:
refactor: padding add, clear filter add

UtpalTheDev committed:
refactor: similar items not showing

UtpalTheDev committed:
refactor: hover color change

UtpalTheDev committed:
Merge pull request #7 from UtpalTheDev/Feature

Feature

UtpalTheDev committed:
refactor: readme update

UtpalTheDev committed:
Merge pull request #8 from UtpalTheDev/Feature

refactor: readme update

UtpalTheDev committed:
refactor: api link update

UtpalTheDev committed:
style update

UtpalTheDev committed:
Merge pull request #10 from UtpalTheDev/Version2

style update

UtpalTheDev committed:
refactor: modified with new version

UtpalTheDev committed:
refactor:modified
